utils/trade.py

```diff
@@ -92,6 +93,26 @@ def get_leagues() -> Tuple[str, ...]:
     return tuple(x["id"] for x in leagues["result"])
 
 
+def get_item_modifiers_by_text() -> dict:
+    global mod_list_dict_text
+    if mod_list_dict_text:
+        return mod_list_dict_text
+    else:
+        item_modifiers = get_item_modifiers()
+        mod_list_dict_text = {item_modifiers[i]["id"]: i for i in range(len(item_modifiers))}
```

Comment:
If this is supposed to allow retrieval by text, shouldn't it be `item_modifiers[i]["text"]`? Haven't looked in a while, so feel free to correct me if I'm wrong.

Reply:
I've updated it to actually work id and text are attributes and as text is not unique it cant be used in as a dict key so we use type and text